src/androidTest/java/com/mixpanel/android/mpmetrics/MixpanelNotificationBuilderTest.java

```diff
@@ -200,6 +248,116 @@ public void testNoActionButtons() {
         verify(builderSpy, never()).addAction(any(Notification.Action.class));
     }
 
+    public void testValidNotificationBadge() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        intent.putExtra("mp_bdgcnt", 2);
+        mpPushSpy.createNotification(intent);
+        verify(builderSpy).setNumber(2);
+    }
+
+    public void testInvalidNotificationBadge() {
+        final Intent intent = new Intent();
+        intent.putExtra("mp_message", "MESSAGE");
+        intent.putExtra("mp_bdgcnt", "0");
```

Comment:
it's unclear whether this is invalid because 0 is not valid or because it's a string. I'd recommend being a bit more specific here or potentially having two tests if both are invalid to make it clear what our expectations are.

Reply:
Invalid because 0 doesn't add any change to the notification; should've been an integer